src/routes/auth/decorators/auth.decorator.ts

```diff
@@ -0,0 +1,13 @@
+import { createParamDecorator, ExecutionContext } from '@nestjs/common';
+import { Request } from 'express';
+import { AuthPayload } from '@/domain/auth/entities/auth-payload.entity';
+
+/**
+ * Route decorator that extracts the {@link AuthPayload}
+ */
+export const Auth = createParamDecorator(
+  (_: unknown, ctx: ExecutionContext): AuthPayload | undefined => {
+    const request: Request = ctx.switchToHttp().getRequest();
+    return request.auth;
+  },
+);
```

Comment:
I'd create a test for this. This can be done via a controller made specifically for testing.

Reply:
Added a test in f047089a1153d5234508a9c1b4d45e85255cd483.